pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -311,6 +311,14 @@ public TimeZone get() {
       types.addCoreType("xml", Oid.XML, Types.SQLXML, "java.sql.SQLXML", Oid.XML_ARRAY);
     }
 
+    if (haveMinimumServerVersion(ServerVersion.v9_2)) {
+      types.addCoreType("json", Oid.JSON, Types.JAVA_OBJECT, "org.postgresql.util.PGobject", Oid.JSON_ARRAY);
```

Comment:
Why do you use `Types.JAVA_OBJECT` here?
It looks like it was declared as `Types.OTHER`.

Reply:
`JAVA_OBJECT`, while not entirely correct is more meaningful than `OTHER`, and it makes no difference either way. But I don't have old server instances running to be able to test the conditional adding. I simply repeated the pattern of the other checks with the appropriate version.

Reply:
The PR is tested via Travis, and it does use multiple PostgreSQL versions (see https://travis-ci.org/pgjdbc/pgjdbc/builds/340856695?utm_source=github_status&utm_medium=notification )

> `JAVA_OBJECT`, while not entirely correct is more meaningful than `OTHER`

This change breaks backward compatibility. Is there a reason to do that?

>I simply repeated the pattern of the other checks with the appropriate version.

The changes should be tested otherwise they can be silently reverted.
The tests are useful as code samples for those who want to use `json`/`jsonb`

Reply:
>This change breaks backward compatibility. Is there a reason to do that?

I believe this change is justified, because `OTHER` can literally mean anything, and there is no *standardized* usage in higher-level APIs that would be broken from it. They're all still "cooking".

If I am to add tests for the presence of the two types, should I change the SQL type back to `OTHER`?